cloud/docker.go

```diff
@@ -45,18 +45,18 @@ func (settings *dockerSettings) Validate() error {
 }
 
 func (s *dockerSettings) FromDistroSettings(d distro.Distro, _ string) error {
-	if d.ProviderSettings != nil {
-		if err := mapstructure.Decode(d.ProviderSettings, s); err != nil {
-			return errors.Wrapf(err, "Error decoding params for distro %s: %+v", d.Id, s)
-		}
-	} else if len(d.ProviderSettingsList) != 0 {
+	if len(d.ProviderSettingsList) != 0 {
 		bytes, err := d.ProviderSettingsList[0].MarshalBSON()
 		if err != nil {
 			return errors.Wrap(err, "error marshalling provider setting into bson")
 		}
 		if err := bson.Unmarshal(bytes, s); err != nil {
 			return errors.Wrap(err, "error unmarshalling bson into provider settings")
 		}
+	} else if d.ProviderSettings != nil {
```

Comment:
Does it make sense to take out the else in these? Like are there situations where both will be populated?

Reply:
Yeah there are, this whole thing is wonky and I don't like it

Reply:
I think that means we don't want the "else" right? Like we want both if statements to be checked?

Reply:
No, it's like "if the new structure is available, populate using that, else populating using the legacy"